app/scripts/modules/core/authentication/authentication.initializer.service.ts

```diff
@@ -70,12 +67,23 @@ export class AuthenticationInitializer {
     this.redirectService.redirect(`${this.settings.gateUrl}/auth/redirect?to=${callback}`);
   }
 
+  public openUserRolesModal() {
```

Comment:
I don't think this really belongs here. I know this service manages the logout modal, but I don't think it needs to be aware of the user roles modal. I would move this into the user menu directive.